### PR TITLE
SC-147: Security Review - #350 BondingCurveFndgMgr issues

### PR DESCRIPTION
**Pull request by Zitzak**
## What has been done?
- Implemented the comments which are checked off in the description of the Linear Issue

## Please note
- Still waiting on reply of reviewers for some of the BC related comments, so I moved them to a [new issue](https://linear.app/inverter/issue/SC-155/security-review-pr350-bonding-curve-funding-manager-2), which will be addressed after their reply
## Implements
- https://linear.app/inverter/issue/SC-147/security-review-pr350-bonding-curve-funding-manager

### CODE REVIEW

File: test/modules/fundingManager/bondingCurveFundingManager/BancorVirtualSupplyBondingCurveFundingManager.t.sol [hunk 1]
```diff
@@ -740,6 +740,7 @@ contract BancorVirtualSupplyBondingCurveFundingManagerTest is ModuleTest {
         uint feeAmount =
             (formulaReturn * fee) / bondingCurveFundingManager.call_BPS();
         uint sellAmountMinusFee = formulaReturn - feeAmount;
+        uint sellAmountPlusFee = feeAmount + sellAmountMinusFee;
```
Comment (FHieser) [inline test/modules/fundingManager/bondingCurveFundingManager/BancorVirtualSupplyBondingCurveFundingManager.t.sol:743]: isnt this just the formulaReturn?

File: test/modules/fundingManager/bondingCurveFundingManager/BancorVirtualSupplyBondingCurveFundingManager.t.sol [hunk 2]
```diff
@@ -764,7 +765,7 @@ contract BancorVirtualSupplyBondingCurveFundingManagerTest is ModuleTest {
                 true, true, true, true, address(bondingCurveFundingManager)
             );
             emit VirtualCollateralAmountSubtracted(
-                sellAmountMinusFee, newVirtualCollateral - sellAmountMinusFee
+                sellAmountPlusFee, newVirtualCollateral - sellAmountPlusFee
```
Comment (FHieser) [inline test/modules/fundingManager/bondingCurveFundingManager/BancorVirtualSupplyBondingCurveFundingManager.t.sol:768]: In addition to the comment from above:
What is happening here that the test is still running through although changes were made here?

**Review by FHieser — COMMENTED** [review]
First two solutions look good.
The last change doesnt make any sense to me.
What is happening there?

File: test/modules/fundingManager/bondingCurveFundingManager/BancorVirtualSupplyBondingCurveFundingManager.t.sol (hunk 1 above)
Comment (Zitzak) [reply test/modules/fundingManager/bondingCurveFundingManager/BancorVirtualSupplyBondingCurveFundingManager.t.sol:743]: Agree, just created a new var for readability in the test

File: test/modules/fundingManager/bondingCurveFundingManager/BancorVirtualSupplyBondingCurveFundingManager.t.sol (hunk 2 above)
Comment (Zitzak) [reply test/modules/fundingManager/bondingCurveFundingManager/BancorVirtualSupplyBondingCurveFundingManager.t.sol:768]: Unsure what you mean. The emit as well as the asset got updated here in this test

Comment (FHieser) [reply test/modules/fundingManager/bondingCurveFundingManager/BancorVirtualSupplyBondingCurveFundingManager.t.sol:768]: Ah okay I think I got it:
I didnt see that _subVirtualCollateralAmount(redeemAmount + feeAmount) would trigger an event here

**Review by FHieser — APPROVED** [review]

**Pull request merged**